### PR TITLE
Set ServerStatusInTrans flag 

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -102,6 +102,15 @@ func (s *SessionVars) SetStatus(status uint16) {
 	s.Status = status
 }
 
+// SetStatusInTrans sets the status flags about ServerStatusInTrans.
+func (s *SessionVars) SetStatusInTrans(isInTrans bool) {
+	if isInTrans {
+		s.Status = s.Status | mysql.ServerStatusInTrans
```
Comment (qiuyesuifeng) [inline sessionctx/variable/session.go]: Using `s.Status |= xxx`?

File: sessionctx/variable/session.go [hunk 2]
```diff
@@ -102,6 +102,15 @@ func (s *SessionVars) SetStatus(status uint16) {
...
+		return
+	}
+	s.Status = s.Status & (^mysql.ServerStatusInTrans)
```
Comment (qiuyesuifeng) [inline sessionctx/variable/session.go]: Using `s.Status &= xxx`

Comment (zimulala) [inline sessionctx/variable/session.go]: got it.

File: tidb_test.go [hunk 3]
```diff
@@ -513,6 +513,45 @@ func (s *testSessionSuite) TestAutoicommit(c *C) {
 	mustExecSQL(c, se, s.dropDBSQL)
 }
 
+func checkInTrans(c *C, se Session, stmt string, isNil bool, expect uint16) {
```
Comment (qiuyesuifeng) [inline tidb_test.go]: Seems no need to use expect or isNil variable? Use one of them is enough?

File: tidb_test.go [hunk 4]
```diff
@@ -513,6 +513,45 @@ func (s *testSessionSuite) TestAutoicommit(c *C) {
...
+	se := newSession(c, store, s.dbName)
+	mustExecSQL(c, se, "drop table if exists t")
+	c.Assert(se.(*session).txn, IsNil)
+	checkInTrans(c, se, "create table t (id BIGINT PRIMARY KEY AUTO_INCREMENT NOT NULL)", true, 0)
+	checkInTrans(c, se, "insert t values ()", true, 0)
+	checkInTrans(c, se, "begin", false, 1)
+	checkInTrans(c, se, "insert t values ()", false, 1)
+	se.Execute("drop table if exists t;")
+	c.Assert(se.(*session).txn, IsNil)
+	c.Assert(variable.GetSessionVars(se.(*session)).Status&mysql.ServerStatusInTrans, Equals, uint16(1))
```
Comment (qiuyesuifeng) [inline tidb_test.go]: A little confused, here why txn is nil but in ServerStatusInTrans?

**Comment by coocood** [comment]
When we begined a transaction, then executed an DDL statement, that transaction is force commited, so the server status in transaction should be set to false.

File: tidb_test.go (hunk 3 above)
Comment (zimulala) [inline tidb_test.go]: no inevitable relation between them.

**Comment by zimulala** [comment]
got it, my fault.

**Comment by coocood** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**